components/Feature/AddGoal/index.js

```diff
@@ -0,0 +1,69 @@
+import { useState } from 'react';
+import { Button, Checkbox, DateInput, TextInput } from 'components/Form';
+
+const AddGoal = ({ planId, updatePlan }) => {
+  const [text, setGoalText] = useState('');
+  const [targetReviewDate, setTargetReviewDate] = useState({});
+  const [useAsPhp, setUseAsPhp] = useState(false);
+
+  const handleGoalTextChange = async e => {
+    await setGoalText(e.target.value);
+  };
```

Comment:
`setGoalText` doesn't need to be awaited. Same for your other calls to set state.